external_services/example.py

```diff
@@ -0,0 +1,91 @@
+from dataclasses import dataclass
+from typing import List
+
+from config import Config
+from external_services.data import get_data
+from external_services.data import post_data
+
+
+@dataclass
+class Example(object):
+    key: str
+    value: str
+
+    @staticmethod
+    def from_json(data: dict):
+        return Example(
+            key=data.get("key"),
+            value=data.get("value"),
+        )
+
+
+class ExampleError(Exception):
+    """Exception raised for errors in EXAMPLE management
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def __init__(self, message="Sorry, there was a problem, please try later"):
+        self.message = message
+        super().__init__(self.message)
+
+
+class ExampleMethods(Example):
+    @staticmethod
+    def list(**kwargs) -> List[Example]:
+        """
+        List examples from the database
+
+        Args:
+            kwargs (dict): keyword args
+
+        Returns:
+            A list of example objects
+        """
+        url = Config.EXAMPLE_STORE_API_HOST + Config.EXAMPLES_ENDPOINT
+        response = get_data(url, kwargs)
+
+        if response:
+            return [Example.from_json(item) for item in response]
+
+    @staticmethod
+    def get_example(key: str) -> Example:
+        """
+        Get an example from an external service
+        corresponding to a given key
+
+        Args:
+            key (str): The example key
+
+        Returns:
+            Example object or raises an ExampleError if none found
+        """
+        url = Config.EXAMPLE_STORE_API_HOST + Config.EXAMPLE_ENDPOINT
+        params = {"key": key}
+        response = get_data(url, params)
+
+        if response and "key" in response:
+            return Example.from_json(response)
+
+    @staticmethod
+    def create_example(key: str, value: str) -> Example | None:
+        """
+        Create a new example if none exists
+
+        Args:
+            key (str): The example key
+            value (str): The example value
+
+        Returns:
+            Example object or None
+        """
+        url = Config.EXAMPLE_STORE_API_HOST + Config.EXAMPLES_ENDPOINT
+        params = {
+            "key": key,
+            "value": value,
+        }
+        response = post_data(url, params)
+
+        if response and "key" in response:
+            return Example.from_json(response)
```

Comment:
⛏️ is there a check for existing examples in the db?

Reply:
So these examples are intentionally bare bones just to create something to test that gets/posts data. I've made it a bit confusing by calling this model example as well as the native api model, but this external_services/example is theoretically a completely different model on a third-party api that you can post to and returns a response (with whatever logic it has) - this is not posting to the connexion api on this repo (that also happens to be called example)!